metrics/test/features/dashboard/presentation/state/project_metrics_store_test.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:metrics/features/dashboard/domain/usecases/parameters/project_id_param.dart';
+import 'package:metrics/features/dashboard/presentation/state/project_metrics_store.dart';
+import 'package:test/test.dart';
+
+void main() {
+  const projectId = 'projectId';
+  const projectIdParam = ProjectIdParam(projectId: projectId);
+  const GetCoverageTestbed getCoverage = GetCoverageTestbed();
+  const GetBuildMetricsTestbed getBuildMetrics = GetBuildMetricsTestbed();
+
+  ProjectMetricsStore projectMetricsStore;
+
+  setUp(() {
+    projectMetricsStore = ProjectMetricsStore(
+      getCoverage,
+      getBuildMetrics,
+    );
+  });
+
+  group("Metrics store creation", () {
+    test("Throws an assert if one of the use cases is null", () {
+      final assertionMatcher = throwsA(const TypeMatcher<AssertionError>());
+
+      expect(() => ProjectMetricsStore(null, null), assertionMatcher);
+      expect(
+        () => ProjectMetricsStore(const GetCoverageTestbed(), null),
+        assertionMatcher,
+      );
+      expect(
+        () => ProjectMetricsStore(null, const GetBuildMetricsTestbed()),
+        assertionMatcher,
+      );
+    });
+  });
+
+  group("Coverage data loading", () {
+    test("Properly loads the coverage data", () async {
+      final actualCoverage = await getCoverage(projectIdParam);
+
+      await projectMetricsStore.getCoverage(projectId);
+
+      expect(actualCoverage, equals(projectMetricsStore.coverage));
+    });
+  });
+
+  group("Builds data loading", () {
+    List<Build> actualBuildMetrics;
+    Build actualFirstBuild;
+    DateTime actualFirstBuildStart;
+
+    setUpAll(() async {
+      actualBuildMetrics = await getBuildMetrics(projectIdParam);
+      actualFirstBuild = actualBuildMetrics.first;
+      actualFirstBuildStart = actualFirstBuild.startedAt;
+    });
+
+    DateTime _trimToDay(DateTime actualFirstBuildStart) {
+      return DateTime(
+        actualFirstBuildStart.year,
+        actualFirstBuildStart.month,
+        actualFirstBuildStart.day,
+      );
+    }
+
+    test("Properly loads the performance points", () async {
+      await projectMetricsStore.getBuildMetrics(projectId);
+
+      final firstPerformancePoint =
+          projectMetricsStore.projectPerformanceMetric.first;
+
+      expect(
+        actualBuildMetrics.length,
+        projectMetricsStore.projectPerformanceMetric.length,
+      );
+      expect(
+        actualFirstBuildStart.millisecondsSinceEpoch,
+        firstPerformancePoint.x,
+      );
+      expect(
+        actualFirstBuild.duration.inMilliseconds,
+        firstPerformancePoint.y,
+      );
+    });
+
+    test("Properly loads the build points", () async {
+      final actualFirstBuildDay = _trimToDay(actualFirstBuildStart);
+
+      final actualFirstBuildCount = actualBuildMetrics
+          .where((build) => _trimToDay(build.startedAt) == actualFirstBuildDay)
+          .length;
+
+      final actualAverageBuildTime = (actualBuildMetrics
```

Comment:
Discussed possibility of moving more logic to the use case and making store more lean that takes care of presentation with Vova offline. Looking forward to the changes.